Nekopost - Search Error "HTTP 404" Fix

keegang6705 opened the pull request:
Closes #9549 

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [ ] Have removed `web_hi_res_512.png` when adding a new extension

keegang6705 committed:
fiex url and request model

keegang6705 committed:
fix request model

keegang6705 committed:
Merge branch 'nekopost-search-error-fix' of https://github.com/keegang6705/extensions-source into nekopost-search-error-fix

keegang6705 committed:
it's working so i made a check point

keegang6705 committed:
everything working fine I've tested it

vetleledaal commented on src/th/nekopost/src/eu/kanade/tachiyomi/extension/th/nekopost/model/SearchRequest.kt:
```diff
@@ -2,8 +2,15 @@ package eu.kanade.tachiyomi.extension.th.nekopost.model
 import kotlinx.serialization.Serializable
 
+@Serializable
+data class PagingRequest(
```
Consider naming this something else. Currently it sounds like a standalone request, not a subset of another request.

vetleledaal commented on src/th/nekopost/src/eu/kanade/tachiyomi/extension/th/nekopost/model/RawProjectSearchSummary.kt:
```diff
@@ -1,19 +1,42 @@
+    val coverVersion: Int = 0,
+    val concatCate: String = "",
+    val editorId: Int = 0,
+    val editorName: String = "",
```
Don't assign a default for fields that are always sent.

Consider omitting fields you aren't using in the code.

keegang6705 committed:
rename PagingRequest to PagingInfo, remove default data from model

keegang6705 committed:
Merge branch 'main' into nekopost-search-error-fix

keegang6705 committed:
use @SerialName for ProjectDate

keegang6705 committed:
Merge branch 'nekopost-search-error-fix' of https://github.com/keegang6705/extensions-source into nekopost-search-error-fix

keegang6705 committed:
use filterNot instead of filter {!...}

keegang6705 committed:
refactor: optimize code

- Use lazy initialization for SimpleDateFormat to avoid repeated instantiation
- Simplify property declarations with type inference
- Replace verbose null checks with Elvis operators and safe calls
- Eliminate unnecessary intermediate variables
- Combine existingProject.add() with map operation for better efficiency
- Inline object creation in SearchRequest constructor
- Use run block for cleaner null handling in popularMangaParse
- Reduce string concatenation by caching basePath in pageListParse
- Remove redundant status checks and variable assignments

tested BUILD SUCCESSFUL
tested fonctionality in mihon
